### PR TITLE
Update experimental Button focus ring to use outline

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Relates to https://github.com/Shopify/web/issues/93863

### WHAT is this pull request doing?

- Update focus ring styles to use `outline` instead of focus ring mixin
- No visual changes

### How to 🎩

- View [Buttons in Storybook](https://5d559397bae39100201eedc1-aauroiygtp.chromatic.com/?path=/story/all-components-button--all&globals=polarisSummerEditions2023:true)
- Activate focus states and ensure no visual regressions

### CODE REVIEW

**Review by jesstelford — APPROVED** [review]

**Review by sophschneider — COMMENTED** [review]
It won't let me approve after merge but it LGTM! Thanks for doing this :D I also am adding this PR to the [se23 issue](https://github.com/Shopify/web/issues/93863) and closing it 🙌

**Pull request merged**
